Makefile

```diff
@@ -42,125 +26,85 @@ ifdef HTTPS_PROXY
 	DOCKERARGS += --build-arg https_proxy=$(HTTPS_PROXY)
 endif
 
-LDFLAGS=
+GO_BUILD_OPTS ?=
+GO_LDFLAGS ?=
+GO_FLAGS ?=
+
 ifdef STATIC
-	export CGO_ENABLED=0
-	LDFLAGS=-a -ldflags '-extldflags \"-static\"'
+	GO_BUILD_OPTS+= CGO_ENABLED=0
+	GO_LDFLAGS+= -extldflags \"-static\"
+	GO_FLAGS+= -a
 endif
 
-export GOPATH
-export GOBIN
-
 V = 0
 Q = $(if $(filter 1,$V),,@)
 
 .PHONY: all
-all: fmt lint build
+all: lint build test
 
-$(BASE): ; $(info  Setting GOPATH...)
-	@mkdir -p $(dir $@)
-	@ln -sf $(CURDIR) $@
+$(BINDIR): ; $(info Creating bin directory...)
+	$Q mkdir -p $@
 
-$(GOBIN):
-	@mkdir -p $@
+$(BUILDDIR): ; $(info Creating build directory...)
+	$Q mkdir -p $@
 
-$(BUILDDIR): | $(BASE) ; $(info Creating build directory...)
-	@cd $(BASE) && mkdir -p $@
+$(COVERAGE_DIR): ; $(info Creating coverage directory...)
+	$Q mkdir -p $@
 
```

Comment:
```
.PHONY: create-dirs
create-dirs: $(BINDIR) $(BUILDDIR) $(COVERAGE_DIR)

$(BINDIR) $(BUILDDIR) $(COVERAGE_DIR):
	$Q mkdir -p $@

```

Reply:
not sure if needed, but might be a nice way to have it...

Reply:
ack, will modify